File: ws/chicaiza/u1/ws06_Json/ChickenFarmSimulatorPython/model/ChickenFarmer.py
from model.ChickenCoop import ChickenCoop

class ChickenFarmer:
    def __init__(self, name: str):
        self.name = name
        self.coops = []

    def add_coop(self, coop: ChickenCoop):
        self.coops.append(coop)

    def to_dict(self):
        return {
            "name": self.name,
            "coops": [c.to_dict() for c in self.coops]
        }

File: hw/ramos/u1/hw14SafeInJsonChickenFarmSimulator/java/ChickenFarmSimulator/src/ec/edu/espe/ChickenFarmSimulator/model/ChickenFarmer.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.ChickenFarmSimulator.model;

/**
 *
 * @author Paulo Ramos
 */
import java.util.ArrayList;
import java.util.List;

public class ChickenFarmer {
    private String name;
    private List<ChickenCoop> coops;

    // Constructor
    public ChickenFarmer(String name) {
        this.name = name;
        this.coops = new ArrayList<>();
    }

    // Agregar un gallinero
    public void addCoop(ChickenCoop coop) {
        coops.add(coop);
    }

    // Obtener todos los pollos de todos los gallineros
    public List<Chicken> getAllChickens() {
        List<Chicken> allChickens = new ArrayList<>();
        for (ChickenCoop coop : coops) {
            allChickens.addAll(coop.getChickens());
        }
        return allChickens;
    }

    // Getter y Setter para el nombre
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
}


